### PR TITLE
[GFX-828] Add checkerboard grays, gradient settings uniforms to Filament's skybox

### PR DESCRIPTION
**Pull request by ViktorHeisenberger**
## Jira ticket URL (Why?)
[GFX-828](https://shapr3d.atlassian.net/browse/GFX-828)

## Short description (What? How?) 📖
We need some iterations on the looks of the gradient and checkerboard pattern backgrounds. Adding uniform variables allows us to iterate on them without needing Filament bumps in the Shapr repository.

### Design review 🎨
N/A

### Affected areas 🧭
Filament's skybox material

### Special use-cases to test 🧷
N/A

### How did you test it? 🤔

Manual 💁‍♂️
Looked at gltf viewer by hand.

### CODE REVIEW

File: filament/src/Skybox.cpp [hunk 1]
```diff
@@ -47,6 +47,8 @@ struct Skybox::BuilderDetails {
     SkyboxType mType = Skybox::SkyboxType::ENVIRONMENT;
     float mUiScale = 1.0f;
     UpDirectionAxis mUpDirectionAxis = UpDirectionAxis::Y_UP;
+    float2 mCheckerboardGrays{1.0, 0.8};
+    float4 mGradientSettings{0.8, 0.93, 0.55, 0.4};
```
Comment (gaborvalasek) [inline filament/src/Skybox.cpp:51]: Relatively nit, but I'd add the f postfix after the float literals.

File: filament/src/materials/skybox.mat [hunk 2]
```diff
@@ -57,8 +65,8 @@ fragment {
     }
 
     float3 shaprCheckerboardPattern(float2 positionScaledClipSpace) {
-        const float colorLight = 1.0;
-        const float colorDark = 0.53;
+        float colorLight = materialParams.checkerboardGrays.x;
```
Comment (gaborvalasek) [inline filament/src/materials/skybox.mat:68]: I'm OK with this for now, however, I'd rather had these parameters constant folded into the shader code once tweaking is done. At least if I'm understanding it correctly, that the gradient curves are not needed to be changed after finetuning with Alex is done.

**Review by gaborvalasek — APPROVED** [review]

File: filament/src/materials/skybox.mat (hunk 2 above)
Comment (ViktorHeisenberger) [reply filament/src/materials/skybox.mat:68]: I agree. Once we set out to do shader variations, we should do it for environments too. The checkerboard will need 2 variants for the public release: light and dark mode.

File: filament/src/Skybox.cpp (hunk 1 above)
Comment (ViktorHeisenberger) [reply filament/src/Skybox.cpp]: Fixed in https://github.com/shapr3d/filament/pull/14/commits/c26d1a69c98809f22d4eb15d018d8c8c45b79777

File: filament/include/filament/Skybox.h [hunk 3]
```diff
@@ -159,6 +159,20 @@ class UTILS_PUBLIC Skybox : public FilamentAPI {
...
+        */
+        Builder& checkerboardGrays(math::float2 grays) noexcept;
+
+        /**
+        * Sets the gradient's position and dimming factor. Dimming factor is in X, upper limit
+        * is in Y, location is in Z, and lower limit is in W.
+        * 
+        * Defaults are 0.8, 0.93, 0.55, 0.4.
+        */
+        Builder& gradientSettings(math::float4 settings) noexcept;
```
Comment (palver123) [inline filament/include/filament/Skybox.h:174]: I'd use a struct with named fields instead of encoding it in a float4. Filament uses such structurs often

**Pull request merged**